### PR TITLE
fixes

### PR DESCRIPTION
**Pull request by screret**
fix unification being broken as shit
fix oregen crashing in dimensions with 0 ore veins
make unification data reset on reload
etc

### CODE REVIEW

File: common/src/main/java/com/gregtechceu/gtceu/api/data/worldgen/ores/OreGenerator.java [hunk 1]
```diff
@@ -82,6 +88,12 @@ private static Optional<BlockPos> computeVeinOrigin(WorldGenLevel level, ChunkGe
         ).findFirst();
     }
 
+    @Nullable
+    public GTOreDefinition getEntry(WorldGenLevel level, Holder<Biome> biome, RandomSource random) {
+        var veins = WorldGeneratorUtils.getCachedBiomeVeins(level.getLevel(), biome, random);
+        int randomEntryIndex = GTUtil.getRandomItem(random, veins, veins.size());
+        return randomEntryIndex == -1 ? null : veins.get(randomEntryIndex).getValue();
+    }
```
Comment (mikerooni) [inline common/src/main/java/com/gregtechceu/gtceu/api/data/worldgen/ores/OreGenerator.java]: Is this supposed to be here? it's not called anywhere.

File: common/src/main/java/com/gregtechceu/gtceu/common/data/GTOres.java [hunk 2]
```diff
@@ -577,7 +584,10 @@ public class GTOres {
                     .parent();
 
     private static GTOreDefinition create(String name, int clusterSize, float density, int weight, WorldGenLayers layer, Supplier<HolderSet<DimensionType>> dimensionFilter, HeightRangePlacement range) {
-        return new GTOreDefinition(GTCEu.id(name), clusterSize, density, weight, layer, dimensionFilter, range, 0.0F, null, null, null);
+        ResourceLocation id = GTCEu.id(name);
+        GTOreDefinition def = new GTOreDefinition(id, clusterSize, density, weight, layer, dimensionFilter, range, 0.0F, null, null, null);;
```
Comment (mikerooni) [inline common/src/main/java/com/gregtechceu/gtceu/common/data/GTOres.java]:
```suggestion
        GTOreDefinition def = new GTOreDefinition(id, clusterSize, density, weight, layer, dimensionFilter, range, 0.0F, null, null, null);
```

**Review by mikerooni — CHANGES_REQUESTED** [review]

File: common/src/main/java/com/gregtechceu/gtceu/api/data/worldgen/ores/OreGenerator.java (hunk 1 above)
Comment (screret) [reply common/src/main/java/com/gregtechceu/gtceu/api/data/worldgen/ores/OreGenerator.java]: Uh i forgot it there

Comment (mikerooni) [reply common/src/main/java/com/gregtechceu/gtceu/api/data/worldgen/ores/OreGenerator.java]:
```suggestion
```

**Review by mikerooni — APPROVED** [review]

**Pull request merged**